### PR TITLE
Added Mongo expression support

### PR DESCRIPTION
**Pull request by sohanlal311**
**1. Issue Link:** 
#26 

**2. Brief explanation of a change:** 
Added support to create Mongo criteria instance to be able to use graphQL generic filters with MongoDB.

**3. Will it break existing clients and code in production?**
No

### CODE REVIEW

**Comment by sohanlal311** [comment]
@jaiswalsk Could you please review this PR?

**Comment by jaiswalsk** [comment]
Thank you @sohanlal311 for your contributions, we will review this and share our feedback soon.

File: src/main/java/com/intuit/graphql/filter/client/ExpressionVisitorFactory.java [hunk 1]
```diff
@@ -15,10 +15,7 @@
  */
 package com.intuit.graphql.filter.client;
 
-import com.intuit.graphql.filter.visitors.ExpressionVisitor;
-import com.intuit.graphql.filter.visitors.InfixExpressionVisitor;
-import com.intuit.graphql.filter.visitors.JpaSpecificationExpressionVisitor;
-import com.intuit.graphql.filter.visitors.SQLExpressionVisitor;
+import com.intuit.graphql.filter.visitors.*;
```
Comment (jaiswalsk) [inline src/main/java/com/intuit/graphql/filter/client/ExpressionVisitorFactory.java:18]: Please replace * import with specific import statements.

File: src/main/java/com/intuit/graphql/filter/visitors/MongoCriteriaExpressionVisitor.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.intuit.graphql.filter.visitors;
+
+import com.intuit.graphql.filter.ast.*;
```
Comment (jaiswalsk) [inline src/main/java/com/intuit/graphql/filter/visitors/MongoCriteriaExpressionVisitor.java:18]: Lets replace the * import with specific import statement.

File: src/test/java/com/intuit/graphql/filter/common/EmployeeDataFetcher.java [hunk 3]
```diff
@@ -96,4 +113,9 @@ public String getSqlExpression() {
     public Specification<Object> getSpecification() {
         return specification;
     }
+
+    public Criteria getCriteria() {
```
Comment (jaiswalsk) [inline src/test/java/com/intuit/graphql/filter/common/EmployeeDataFetcher.java:117]: Its better to rename this method as `getMongoCriteria()`

**Review by jaiswalsk — CHANGES_REQUESTED** [review]

File: src/main/java/com/intuit/graphql/filter/visitors/MongoCriteriaExpressionVisitor.java (hunk 2 above)
Comment (sohanlal311) [reply src/main/java/com/intuit/graphql/filter/visitors/MongoCriteriaExpressionVisitor.java]: Its done.

File: src/test/java/com/intuit/graphql/filter/common/EmployeeDataFetcher.java (hunk 3 above)
Comment (sohanlal311) [reply src/test/java/com/intuit/graphql/filter/common/EmployeeDataFetcher.java]: Its done.

**Comment by sohanlal311** [comment]
@jaiswalsk Thanks for your review. I have done the changes and also rebased my branch with the latest master. Please review.

**Review by jaiswalsk — APPROVED** [review]
Looks good!

**Pull request merged**